lightnode/kvstore/storeutil/ttl_wrapper.go

```diff
@@ -0,0 +1,272 @@
+package storeutil
+
+import (
+	"context"
+	"github.com/Layr-Labs/eigenda/lightnode/kvstore"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"time"
+)
+
+var _ kvstore.Store = &ttlStore{}
+
+// ttlStore adds a time-to-live (TTL) capability to the store.
+//
+// This store utilizes the properties of store iteration. Namely, that the keys are returned in lexicographical order,
+// as well as the ability to filter keys by prefix. "Regular" keys are stored in the store with a prefix "k", while
+// special expiry keys are stored with a prefix "e". The expiry key also contains the expiry time in hexadecimal format,
+// such that when iterating over expiry keys in lexicographical order, the keys are ordered by expiry time. The value
+// each expiry key points to is the regular key that is to be deleted when the expiry time is reached. In order to
+// efficiently delete expired keys, the expiry keys must be iterated over periodically to find and delete expired keys.
+type ttlStore struct {
+	store  kvstore.Store
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	logger logging.Logger
+}
+
+// TTLWrapper extends the given store with TTL capabilities. Periodically checks for expired keys and deletes them
+// with a period of gcPeriod. If gcPeriod is 0, no background goroutine is spawned to check for expired keys.
+func TTLWrapper(
+	ctx context.Context,
+	logger logging.Logger,
+	store kvstore.Store,
+	gcPeriod time.Duration) kvstore.TTLStore {
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	ttlStore := &ttlStore{
+		store:  store,
+		ctx:    ctx,
+		cancel: cancel,
+		logger: logger,
+	}
+	if gcPeriod > 0 {
+		ttlStore.expireKeysInBackground(gcPeriod)
+	}
+	return ttlStore
+}
+
+var keyPrefix = []byte("k")
+var expiryPrefix = []byte("e")
```

Comment:
What if they collide with normal keys?

Reply:
Short answer, if used properly it's impossible to get key collisions.

If somebody is using the TTL wrapper, then it is unsafe to access the store directly. If somebody only accesses through the TTL wrapper, then every key gets a prefix, meaning it is impossible for there to be collisions. All "normal" keys are prefixed with "k", all expiry keys are prefixed with "e".

I'll add a comment pointing out that it's unsafe to access the store directly if you are also using the TTL wrapper.

Reply:
```
// Note: it is unsafe to access the wrapped store directly while the TTLStore is in use. The TTLStore uses special
// key formatting, and direct access to the wrapped store may violate the TTLStore's invariants, resulting in
// undefined behavior.
```

Reply:
Sounds good to me to have the wrapper to handle the namespacing